cli/cmd/inject.go

```diff
@@ -30,11 +30,6 @@ const (
 	udpDesc            = "pod specs do not include UDP ports"
 )
 
-type injectOptions struct {
-	disableIdentity bool
-	*proxyConfigOptions
-}
-
```

Comment:
`--disable-identity` will be an option shared by both `linkerd inject` and `linkerd install`, so I moved it into `proxyConfigOptions` and we no longer need a separate `injectOptions` struct.

Reply:
Mmm... I'm not sure that this would be safe to apply on upgrade. Almost certainly would produce a broken cluster.

Reply:
`linkerd upgrade --disable-identity` is not allowed. Or you meant something else?

Reply:
Oh, I'm confusing things. As long as this doesn't cause disable-identity to be included in the upgrade help output, I'm happy.

Reply:
To confirm the implication behind adding `--disable-identity` to `install`, in this set up,

1. The identity service is still installed in the control plane
1. By default, all meshed services will not participate in identity
1. The only way for any meshed services to participate in identity is via the new `config.linkerd.io/disable-identity=false` annotation
1. It won't be possible to enable global identity again during upgrade, because it's not a supported flag at the moment.

Am I right?

Reply:
One thing to call out here -- if control plane proxies are not instrumented with identity, then no other pods will be able to use identity.  So `--disable-identity` at install-time should probably set a flag on the proxy config to configure the default injection policy, but we probably should avoid applying this policy to the control plane itself.

Reply:
@ihcsim @olix0r Those are great points, thanks. I'll keep them in mind when doing the install part.
Adding a new flag for the default injection policy will facilitate re-enabling it in `linkerd update`, if we want to implement that at some point.

Reply:
@alpeb and an alternative, also viable approach, would be to let `install --disable-identity` disable CP proxy identity; and if `disable-identity: false` is specified, injection would fail.

cc @grampelberg